### DISCUSSION TITLE
Upcoming release 2.2.0 & PHP requirements

### DISCUSSION THREAD

**Discussion by Levdbas** [timber/timber#2920] [Announcements]
Hi community,

We are busy with collecting the final tickets for Timber 2.1.0 where we will squash the most prominent bugs that have been reported.
After that, we are going to focus on Timber version 2.2.0. where we will bump the minimal required PHP version to 8.1

So this is a public service announcement so you can prepare your development machines and server for this upcoming change.

**Comment by szepeviktor** [comment]
The WordPress ecosystem is getting ready for PHP 8: https://johnbillion.github.io/wp-stats/php.html

**Reply by Levdbas** [reply]
Yes, but the majority of the WP installs still runs on PHP7.4. But our userbase of Timber 2.0 have composer installed, our thinking goes that it wouldn't be that difficult for them to update to a higher PHP version if not already on a modern version. Looking at our install stats at packagist we see that rougly 20% still use PHP 8.0 and lower. Still a big percentage, but better then the 49% globally when looking at your linked stats.

How do you look at that @szepeviktor ?

**Reply by szepeviktor** [reply]
I understand your point.

Let's say I [install WP core + Timber](https://github.com/szepeviktor/composer-managed-wordpress) on PHP 8, that is okay.
How about installing a random plugin? This is why PHP 7.4 should be used. Otherwise many plugins will have some problems as they are not widely tested/used on PHP 8.

**Reply by szepeviktor** [reply]
My policy in 2024 is to run WordPress on PHP 7.4.
Maybe at the end of the year or in 2025 a PHP 8 version could be used.

One of the legs of WP core ("infinite BC") is still in PHP 4!

**Comment by rmens** [comment]
Running Timber 2.0.0 on PHP 8.3 without problems. Modernizing the code for PHP 8.1 sounds like a great idea!
